### ISSUE TITLE
CloudFormation template unique reources

### ISSUE THREAD

**Issue by amirbenun** [elastic/cloudbeat#840]

**Comment by eyalkraft** [comment]
From the daily:

- [ ] Make sure - Does this problem mean users can't install the cloudformation on two different regions?
- [ ] If so, This is priority 0
- [ ] When fixing, Product input is required - Do we want the cloudformation to fail when installed on the same region twice? @nick-alayil

**Comment by amirbenun** [comment]
After validating, we create an `AWS::IAM::InstanceProfile`, which we then assign to an EC2.
I guess this object is not scoped per region as we can't create two of them with the same name even on different regions.
I am changing this task to priority 0, it has a very quick fix, and I will shoot a pr today/tomorrow.

**Comment by nick-alayil** [comment]
> Does this problem mean users can't install the cloudformation on two different regions?

Certainly, user must be able to install cloudformation on two different regions.

> When fixing, Product input is required - Do we want the cloudformation to fail when installed on the same region twice?

Ideally, repeated run of a cloudformation on already installed region should fail. If it's a tedious and complex fix, we can mention it as a known issue(warning msg) in the documentation, but @amirbenun's above message sounds like we're good with the fix.

**Comment by amirbenun** [comment]
Not exactly, @nick-alayil.
With my PR, users will be able to deploy the stack more than once in the same region as long they will use unique naming.
However, the template already contains a resource of type `AWS::EC2::SecurityGroup` whose name should be unique within the region. Giving the Security Group a constant name (that does not depend on the unique id) will result in blocking the user from provisioning the stack more than once on the same region, but will still allow provisioning more than once on different regions.
I can't say I like this implicit solution, as it is not clear but it achieves the goal.

**Comment by olegsu** [comment]
Can we tell a user upon installation that we see that there is an active instance running on this region (as p1, or course)? 
so duplication might cause unexpected behavior and costs

**Comment by tehilashn** [comment]
@olegsu - this is one possible solution but it's not bulletproof. 
I think that what we have at the moment is good enough for the beta release.

The desired behaviour should be defined by @nick-alayil  and tracked on different tickets.

**Comment by kfirpeled** [comment]
QA: check that deploying with Cloudformation on two different regions works and produce vulnerabilities

**Comment by uri-weisman** [comment]
verified by deploying the cloudformation stack on two different regions.

![Image](https://github.com/elastic/cloudbeat/assets/68195305/92e630a4-c6eb-4350-ab25-a921cc1f4c5e)
